proxy/ptunnel/commands/agent.go

```diff
@@ -5,37 +5,43 @@ import (
 	"github.com/enfabrica/enkit/lib/client"
 	"github.com/enfabrica/enkit/lib/kcerts"
 	"github.com/spf13/cobra"
+	"os"
+	"os/exec"
+	"strings"
 )
 
 type agentConfig struct {
 	Print          bool
 	ListIdentities bool
+	RemainingArgs  []string
+	UseHostEnv     bool
 }
 
 func NewAgentCommand(bf *client.BaseFlags) *cobra.Command {
 	agentConfig := &agentConfig{}
 	c := &cobra.Command{
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return RunAgentCommand(bf, agentConfig)
+			agentConfig.RemainingArgs = args
+			return RunAgentCommand(cmd, bf, agentConfig)
 		},
-		Use:   "agent",
-		Short: "commands for the enkit specific ssh-agent",
+		Use:   "agent -- [Command]",
+		Short: "commands for the enkit specific ssh-agent, anything passed in will execute with SSH_AUTH_SOCK and SSH_AGENT_PID set for the enkti agent.",
 	}
 	// Note the following is intended to be user friendly, identities here are cert principals
 	c.Flags().BoolVarP(&agentConfig.ListIdentities, "list-identities", "l", false, "list the identities loaded current in the agent")
 	c.Flags().BoolVarP(&agentConfig.Print, "print", "p", false, "print the socket and PID of the running agent")
+	c.Flags().BoolVarP(&agentConfig.UseHostEnv, "use-env", "e", false, "Use your terminals env when executing also")
```

Comment:
I would personally remove this option, and always do as if use-env was true. That is generally the expected behavior when forking a command. If a clean environment is needed, one can wrap the command in `/bin/env`

Reply:
Up to you, I don't have an opinion. Implemented